dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/introspection/JacksonPropertyIntrospector.java

```diff
@@ -309,22 +311,12 @@ private static List<Property> collectProperties(Class<?> klass) {
       return Collections.emptyList();
     }
 
-    List<Field> fields =
-        ReflectionUtils.findFields(klass, f -> f.isAnnotationPresent(JsonProperty.class));
-
-    List<Method> methods =
-        ReflectionUtils.findMethods(
-            klass,
-            m ->
-                AnnotationUtils.findAnnotation(m, JsonProperty.class) != null
-                    && m.getParameterTypes().length == 0);
-
-    Multimap<String, Method> multimap = ReflectionUtils.getMethodsMultimap(klass);
```

Comment:
FYI: This map was used to see if there is a setter for the expected name but it is unnecessary complicated and also would just take the first method with a matching name not making sure it is actually a setter with matching type parameter. This now uses `findSetterMethod(fieldName, klass, type)` instead where `type` is the field or getter return type.